### ISSUE TITLE
SpanContextParse Exception with multiple Trace Ids

### ISSUE THREAD

**Issue by gr0l** [inspectIT/inspectit-ocelot#871]
Affected version: 1.1
When the correlation data field X-B3-SpanId contains more than 1 span Id an SpanParseException is thrown:

```
r.i.o.c.i.c.ContextPropagationUtil       : Error reading trace correlation data from B3 headers: ["X-B3-SpanId": "70f6efcb0a279b53,70f6efcb0a279b53", "X-B3-Sampled": "1,1", "X-B3-TraceId": "3787de559048209ce54a06a5a581822b,3787de559048209ce54a06a5a581822b"]
java.lang.IllegalArgumentException: Invalid size: expected 32, got 65
	at io.opencensus.internal.Utils.checkArgument(Utils.java:67)
 	at io.opencensus.trace.TraceId.fromLowerBase16(TraceId.java:106)
 	at io.opencensus.implcore.trace.propagation.B3Format.extract(B3Format.java:92)
 	... 29 common frames omitted
 Wrapped by: io.opencensus.trace.propagation.SpanContextParseException: Invalid input.
 	at io.opencensus.implcore.trace.propagation.B3Format.extract(B3Format.java:110)
 	at rocks.inspectit.ocelot.core.instrumentation.context.ContextPropagationUtil.readPropagatedSpanContextFromHeaderMap(ContextPropagationUtil.java:171)
 	at rocks.inspectit.ocelot.core.instrumentation.context.InspectitContextImpl.readDownPropagationHeaders(InspectitContextImpl.java:472)
 	at de.hybris.platform.servicelayer.web.inspectitGen$$$8154026093234827050.executeImpl(VoidGenericActionTemplate.java)
 	at de.hybris.platform.servicelayer.web.inspectitGen$$$8154026093234827050.execute(VoidGenericActionTemplate.java:23)
 	at rocks.inspectit.ocelot.core.instrumentation.actions.bound.VoidDynamicBoundGenericAction.execute(VoidDynamicBoundGenericAction.java:27)
 	at rocks.inspectit.ocelot.core.instrumentation.hook.MethodHook.onEnter(MethodHook.java:61)
 	at de.hybris.platform.servicelayer.web.XSSFilter.doFilter(XSSFilter.java:298)
 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:609)
 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:810)
 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1623)
 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
 	at io.grpc.Context$1.run(Context.java:595)
 	at io.grpc.Context$1.run(Context.java:595)
 	at io.grpc.Context$1.run(Context.java:595)
 	at io.grpc.Context$1.run(Context.java:595)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by JonasKunz** [comment]
Hi @gr0l ,

The exception shown here is valid because the B3 headers are expected to contain only a single span.

In your case the Headers contains the ID of the same span twice, which indicates that the ID of the same span has been injected two times by accident.

Are you also using inspectIT on the sending side of your request?
If yes, in which version?
We had this issue in the past with some requests sent via HTTPUrlConnection but it should be fixed in the latest release.

**Comment by mariusoe** [comment]
Hi @gr0l 

you wrote you are using version 1.1.
As @JonasKunz already wrote, this bug shoud have been fixed in the latest version 1.4.

For more details see ticket #743 

Please feel free to reopen this issue in case the problem still persists on your side.
